GoogleForCreators/web-stories-wp
# Design System Storybook - Typography

## Summary

- Adds typography atoms from figma to new design system subdirectory with storybook demos and some unit tests 
(Earth section) https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=1286%3A457091

## Relevant Technical Choices

- I sectioned out the typography in three sections: `Display`, `Headline`, and `Text`. `Display` and `Headline` require you to specify what heading level you want, `Text` allows for spans, paragraphs and anchors. We'll handle lists separately. Avoiding complexity at this granular of a level. 
- Constants for sizes to avoid future typos and help developers out with a known source of truth for what values are acceptable off the bat. 

## To-do
- n/a

## User-facing changes

- None, just storybook

## Testing Instructions

- Spin up storybook and see that there's a new section for typography added to the design system section. There should be sections for typography and colors. 
- I added a few unit tests for typography atoms, they're super basic but are a good sanity check for the theme values mapping.

<img width="991" alt="Screen Shot 2020-10-27 at 10 12 14 AM" src="https://user-images.githubusercontent.com/10720454/97337015-f310c900-183c-11eb-9485-3d1ffe729c0a.png">


---



Addresses #4934
